packages/template/src/components-page/cli-auth-confirm.tsx

```diff
@@ -0,0 +1,121 @@
+'use client';
+
+import { Typography } from "@stackframe/stack-ui";
+import { useState } from "react";
+import { useStackApp } from "..";
+import { MessageCard } from "../components/message-cards/message-card";
+import { useTranslation } from "../lib/translations";
+
+export function CLIConfirmation({ fullPage = true }: { fullPage?: boolean }) {
+  const { t } = useTranslation();
+  const app = useStackApp();
+  const [authorizing, setAuthorizing] = useState(false);
+  const [success, setSuccess] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+
+  const user = app.useUser();
+
+  const handleAuthorize = async () => {
+    if (authorizing) return;
+
+    setAuthorizing(true);
+    try {
+      // Get login code from URL query parameters
+      const urlParams = new URLSearchParams(window.location.search);
+      const loginCode = urlParams.get("login_code");
+
+      if (!loginCode) {
+        throw new Error("Missing login code in URL parameters");
+      }
+
+      // We can get a session from the user, which is currently logged in
+      if (!user) {
+        throw new Error("You must be signed in to authorize CLI applications");
+      }
+
+      // Get the refresh token from session storage
+      const refreshTokenStr = localStorage.getItem(`stack.tokens.${app.projectId}.refresh_token`);
+      if (!refreshTokenStr) {
+        throw new Error("No refresh token found. Please sign in again.");
+      }
+
+      try {
+        // Make the fetch request directly
+        const response = await fetch(`${window.location.origin}/api/v1/auth/cli/login`, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+            'X-Stack-Project-Id': app.projectId,
+          },
+          body: JSON.stringify({
+            login_code: loginCode,
+            refresh_token: refreshTokenStr,
+          }),
+        });
+
+        if (!response.ok) {
+          const errorData = await response.json();
+          throw new Error(errorData.message || 'Failed to authorize CLI');
+        }
```

Comment:
This should be more human-readable than just `.message`. Also, in the error case, the response might not be JSON